eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -192,7 +194,9 @@ object Graph {
                                    boundaries: RichWeight => Boolean,
                                    currentBlockHeight: Long,
                                    wr: WeightRatios,
-                                   includeLocalChannelCost: Boolean): Seq[GraphEdge] = {
+                                   includeLocalChannelCost: Boolean,
+                                   firstHopChannelId: Option[ShortChannelId],
+                                   lastHopChannelId: Option[ShortChannelId]): Seq[GraphEdge] = {
```

Comment:
Please don't touch `dijkstraShortestPath`, there is no need. Instead search for a path from the end of `firstHopChannelId` to the start of `lastHopChannelId` and add our `nodeId` to `ignoredVertices`.

Reply:
It turns out there's no need in changing `Router` at all. All above can be done over the API parameters (see https://github.com/ACINQ/eclair/pull/1969). I'm closing this PR.